packages/ii-login-button/src/ii-login-button.ts

```diff
@@ -0,0 +1,277 @@
+ * })
+ * button.addEventListener('login', (e)=>{
+ *  // Access authClient
+ *  const client = e.target.authClient;
+ *  // Principal
+ *  const principal = e.target.principal // also principalString
+ *  // AccountIdentifier for ICP ledger
+ *  const account = e.target.accountIdentifier
+ * })
+ */
+class IILoginButton extends LoginButton {
+  private _state: State;
+  constructor() {
+    super();
+    if (!this.shadowRoot) {
+      this.attachShadow({ mode: "open" });
+    }
+    this._state = {
+      isAuthenticated: false,
+      ready: false,
+    };
+
+    this.init();
+    this.render();
+  }
+
+  private init() {
+    AuthClient.create(this._state.createOptions)
+      .then(async (client) => {
+        this.setState({
+          ...this._state,
+          authClient: client,
+          isAuthenticated: await client.isAuthenticated(),
+          identity: await client.getIdentity(),
+        });
+        // fire ready event
+        if (!this._state.ready) {
+          const event = new ReadyEvent("ready");
+          this.dispatchEvent(event);
+          this.setState({ ready: true });
+        }
+      })
+      .catch((error) => {
+        console.error(error);
+        this.setState({ authClient: undefined });
+      });
+  }
+
+  connectedCallback() {
+    const login = this._login.bind(this);
+    this.shadowRoot
+      ?.querySelector("#ii-login-button")
+      ?.addEventListener("click", login);
+  }
+
+  setState(newState: { [Property in keyof State]+?: State[Property] }) {
+    this._state = {
+      ...this._state,
+      ...newState,
+    };
+  }
+
+  private render() {
+    const shadowRoot = this.shadowRoot as ShadowRoot;
+    const oldButton = shadowRoot.querySelector("button");
+    const newButton = document.createElement("button");
+    newButton.id = "ii-login-button";
+
+    const logoRight = this.hasAttribute("logo-right");
+    const label = this.getAttribute("label") ?? "Login With Internet Identity";
+    if (logoRight) {
+      newButton.innerHTML = `
+          <span slot="label">${label}</span>
+          <span slot="logo">${iiLogo}</span>`;
+    } else {
+      newButton.innerHTML = `
+        <span slot="logo">${iiLogo}</span>
+        <span slot="label">${label}</span>`;
+    }
+
+    if (oldButton) {
+      shadowRoot.removeChild(oldButton);
+    }
+    shadowRoot.appendChild(newButton);
+    const login = this._login.bind(this);
+    newButton.addEventListener("click", login);
+  }
+
+  attributeChangedCallback(attrName: string, oldVal: unknown, newVal: unknown) {
+    const button = this.shadowRoot?.getElementById(
+      "ii-login-button"
+    ) as HTMLButtonElement;
+    switch (attrName) {
+      case "disabled": {
+        button.disabled = newVal === "true";
+        break;
+      }
+      case "label": {
+        (
+          button.querySelector("#ii-login-button-label") as HTMLSpanElement
+        ).innerText = newVal as string;
+        break;
+      }
+      case "innerstyle": {
```

Comment:
If you want to allow consumer to style entirely the button I would sugget the use a CSS `::part` not an attribute